terraform/provider/provider.go

```diff
@@ -1,78 +1,339 @@
 				Description: "Path to Teleport Root CA",
 			},
+			profileName: {
+				Type:        schema.TypeString,
+				DefaultFunc: schema.EnvDefaultFunc("TF_TELEPORT_PROFILE_NAME", ""),
+				Optional:    true,
+				Description: "Teleport profile name",
+			},
+			profileDir: {
+				Type:        schema.TypeString,
+				DefaultFunc: schema.EnvDefaultFunc("TF_TELEPORT_PROFILE_DIR", ""),
+				Optional:    true,
+				Description: "Teleport profile path",
+			},
+			identityFilePath: {
+				Type:        schema.TypeString,
+				DefaultFunc: schema.EnvDefaultFunc("TF_TELEPORT_IDENTITY_FILE_PATH", ""),
+				Optional:    true,
+				Description: "Teleport identity file path",
+			},
 		},
 		ResourcesMap: map[string]*schema.Resource{
-			"teleport_user": resourceTeleportUser(),
-			"teleport_role": resourceTeleportRole(),
+			"teleport_role":             resourceTeleportRole(),
+			"teleport_user":             resourceTeleportUser(),
+			"teleport_github_connector": resourceTeleportGithubConnector(),
+			"teleport_oidc_connector":   resourceTeleportOIDCConnector(),
+			"teleport_saml_connector":   resourceTeleportSAMLConnector(),
+			"teleport_provision_token":  resourceTeleportProvisionToken(),
+			"teleport_trusted_cluster":  resourceTeleportTrustedCluster(),
 		},
-		ConfigureFunc: providerConfigure,
+		DataSourcesMap: map[string]*schema.Resource{
+			"teleport_role":             dataSourceTeleportRole(),
+			"teleport_user":             dataSourceTeleportUser(),
+			"teleport_github_connector": dataSourceGithubConnector(),
+			"teleport_oidc_connector":   dataSourceTeleportOIDCConnector(),
+			"teleport_saml_connector":   dataSourceTeleportSAMLConnector(),
+			"teleport_provision_token":  dataSourceTeleportProvisionToken(),
+			"teleport_trusted_cluster":  dataSourceTeleportTrustedCluster(),
+		},
+		ConfigureContextFunc: configureContext,
 	}
 }
 
-// providerConfigure accepts provider settings schema,
+// configureContext accepts provider settings schema,
 // initializes a Teleport's `auth.Client` and returns it.
-func providerConfigure(d *schema.ResourceData) (interface{}, error) {
-	certPath := d.Get("cert_path").(string)
-	keyPath := d.Get("key_path").(string)
-	rootCAsPath := d.Get("root_ca_path").(string)
+func configureContext(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	config, err := getConfig(d)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+
+	client, err := client.New(ctx, *config)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
 
-	// TODO: Verify Addr format is [protocol://]host:port
-	addr := d.Get("addr").(string)
+	return client, diag.Diagnostics{}
+}
+
+// getConfig loads client config from a given identity source
+func getConfig(d *schema.ResourceData) (*client.Config, error) {
+	_, ok_cert := d.GetOk(certPathKey)
+	_, ok_root_ca := d.GetOk(rootCaPathKey)
+	_, ok_key := d.GetOk(keyPathKey)
+	_, ok_id_file := d.GetOk(identityFilePath)
+
+	switch {
+	case ok_cert || ok_root_ca || ok_key:
```

Comment:
Shouldn't this check that all 3 are set?

Reply:
I think, it should check one of because if user specified one of the values, it means that he wants to use key file auth. We should not proceed to other auth methods.